### PR TITLE
Make CI more dynamic

### PR DESCRIPTION
**Pull request by michal-dagan**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-8349
nightly: https://code.pan.run/xsoar/content/-/pipelines/6542395

### CODE REVIEW

**Review by GuyAfik — APPROVED** [review]
@michal-dagan awesome!

**please add a change-log** before merging**

File: demisto_sdk/commands/common/tests/git_config_test.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def test_search_gitlab_id_valid(self, mocker, requests_mock):
         git_config = GitContentConfig()
         assert (
             git_config.project_id == 3606
-        )  # this is the project id of `content-internal-dist`
+        )  # this is the project id of `content-internal-dist` in code.pan.run
```
Comment (kobymeir) [inline demisto_sdk/commands/common/tests/git_config_test.py:252]:
```suggestion
        )  # this is the project id of `content-internal-dist`
```

File: demisto_sdk/commands/common/update_id_set.py [hunk 2]
```diff
@@ -3793,9 +3794,9 @@ def is_same_source(source1, source2) -> bool:
     host1, owner1, repo1 = source1
     host2, owner2, repo2 = source2
     if (
-        host1 in {"github.com", "code.pan.run"}
+        host1 in {"github.com", "code.pan.run", DEMISTO_SDK_CI_SERVER_HOST}
```
Comment (kobymeir) [inline demisto_sdk/commands/common/update_id_set.py:3797]: why both values? 
```suggestion
        host1 in {"github.com", DEMISTO_SDK_CI_SERVER_HOST}
```

File: demisto_sdk/commands/common/update_id_set.py [hunk 3]
```diff
@@ -3793,9 +3794,9 @@ def is_same_source(source1, source2) -> bool:
...
         and owner1 in {"demisto", "xsoar"}
-        and host2 in {"github.com", "code.pan.run"}
+        and host2 in {"github.com", "code.pan.run", DEMISTO_SDK_CI_SERVER_HOST}
```
Comment (kobymeir) [inline demisto_sdk/commands/common/update_id_set.py:3799]:
```suggestion
        and host2 in {"github.com", DEMISTO_SDK_CI_SERVER_HOST}
```

**Review by kobymeir — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,7 @@
 * Fixed an issue where **validate** falsely failed with error `DS108` on descriptions ending with new lines followed by square/curly brackets.
 * Fixed an issue where **graph** commands would not clean their temporary files properly, causing successive commands to fail.
 * Fixed an issue where an error log message changed the terminal color.
+* Enhanced configurability by introducing more dynamic settings.
```
Comment (GuyAfik) [inline CHANGELOG.md:12]:
```suggestion
* Added the <env_var1>, <env_var2>... environment variables.
```
we should document all those new env vars,
but since this PR is urgent, you can do it in a separate PR.

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,7 @@
 * Fixed an issue where **validate** falsely failed with error `DS108` on descriptions ending with new lines followed by square/curly brackets.
 * Fixed an issue where **graph** commands would not clean their temporary files properly, causing successive commands to fail.
 * Fixed an issue where an error log message changed the terminal color.
+* Updated the code to use the following environment variables: GCS_MARKET_BUCKET, GCS_MARKET_BUCKET_DEV, GCS_MARKET_XPANSE_BUCKET, GCS_MARKET_V2_BUCKET, GCS_MARKET_XSOAR_SAAS_BUCKET, DOCKER_IO, CI_PROJECT_ID, CI_PROJECT_ID.
```
Comment (GuyAfik) [inline CHANGELOG.md:12]:
```suggestion
*Added the `GCS_MARKET_BUCKET`, `GCS_MARKET_BUCKET_DEV`, `GCS_MARKET_XPANSE_BUCKET`, `GCS_MARKET_V2_BUCKET`, `GCS_MARKET_XSOAR_SAAS_BUCKET`, `DOCKER_IO`, `CI_PROJECT_ID`, `CI_PROJECT_ID`.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -9,6 +9,7 @@
 * Fixed an issue where **validate** falsely failed with error `DS108` on descriptions ending with new lines followed by square/curly brackets.
 * Fixed an issue where **graph** commands would not clean their temporary files properly, causing successive commands to fail.
 * Fixed an issue where an error log message changed the terminal color.
+* Added the `GCS_MARKET_BUCKET`, `GCS_MARKET_BUCKET_DEV`, `GCS_MARKET_XPANSE_BUCKET`, `GCS_MARKET_V2_BUCKET`, `GCS_MARKET_XSOAR_SAAS_BUCKET`, `DOCKER_IO`, `CI_PROJECT_ID`, `CI_PROJECT_ID` environment variables.
```
Comment (dorschw) [inline CHANGELOG.md:12]: If any of these are relevant for users, explain what they're for. Otherwise, add the `Internal:` prefix, and a short description for the usages.

File: demisto_sdk/commands/lint/linter.py [hunk 7]
```diff
@@ -365,8 +366,7 @@ def _gather_facts(self, modules: dict) -> bool:
             # we want to use the docker-io.art.code.pan.run only if we run in content build (and not CI/CD for example)
             if os.getenv("CONTENT_GITLAB_CI", False):
                 self._facts["images"] = [
-                    [f"docker-io.art.code.pan.run/{image[0]}", -1]
-                    for image in self._facts["images"]
+                    [f"{DOCKER_IO}/{image[0]}", -1] for image in self._facts["images"]
```
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:369]: make sure to manually test it

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,6 +2,7 @@
 ## Unreleased
 * Fixed an issue where unified integrations / scripts with a period in their name would not split properly.
 * Improved logging for **lint** and **prepare-content** commands.
+* Internal: Added the `GCS_MARKET_BUCKET`, `GCS_MARKET_BUCKET_DEV`, `GCS_MARKET_XPANSE_BUCKET`, `GCS_MARKET_V2_BUCKET`, `GCS_MARKET_XSOAR_SAAS_BUCKET`, `DOCKER_IO`, `CI_PROJECT_ID`, `CI_PROJECT_ID` environment variables.
```
Comment (michal-dagan) [inline CHANGELOG.md:5]:
```suggestion
* Internal: Updated the code to use `DOCKER_IO`, `CI_SERVER_HOST`, `CI_PROJECT_ID` environment variables.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -2,6 +2,7 @@
 ## Unreleased
 * Fixed an issue where unified integrations / scripts with a period in their name would not split properly.
 * Improved logging for **lint** and **prepare-content** commands.
+* Internal: Updated the code to use `DOCKER_IO`, `CI_SERVER_HOST`, `CI_PROJECT_ID` environment variables.
```
Comment (GuyAfik) [inline CHANGELOG.md:5]:
```suggestion
* Internal: Added the `DOCKER_IO`, `CI_SERVER_HOST`, `CI_PROJECT_ID` environment variables.
```

**Review by kobymeir — APPROVED** [review]
